tests/integration/test_load_and_run_multi_agent_a2a_tool.py

```diff
@@ -0,0 +1,150 @@
+                traces[parent_k].attributes[CHILD_TAG][trace.context.span_id] = trace
+            else:
+                traces[None] = trace
+    logger.info(traces[None].model_dump_json(indent=2))
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+def test_load_and_run_multi_agent(
+    agent_framework: AgentFramework,
+    check_multi_tool_usage: Callable[[list[AgentSpan]], None],
+) -> None:
+    kwargs = {}
+
+    if agent_framework is AgentFramework.TINYAGENT:
+        pytest.skip(
+            f"Skipping test for {agent_framework.name} because it does not support multi-agent"
+        )
+
+    kwargs["model_id"] = "gpt-4.1-nano"
+    agent_model = kwargs["model_id"]
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else None
+    )
+
+    search_agent_port = 5800
+    search_agent_endpoint = "search_agent"
+    search_agent_description = "Agent that can search the web. It can find answers on the web if the query cannot be answered."
+
+    search_agent_cfg = AgentConfig(
+        instructions="Use the available tools to complete the task to obtain additional information to answer the query.",
+        name="search_web_agent",
+        model_id=agent_model,
+        description=search_agent_description,
+        tools=[search_web],
+        model_args=model_args,
+    )
+
+    main_agent_cfg = AgentConfig(
+        instructions="Use the available tools to complete the task to obtain additional information to answer the query.",
+        description="The orchestrator that can use other agents via tools using the A2A protocol.",
+        tools=[a2a_query(f"http://localhost:{search_agent_port}/{search_agent_endpoint}", search_agent_cfg.description)],
+        model_args=model_args,
+        **kwargs,  # type: ignore[arg-type]
+    )
+
+    search_agent = AnyAgent.create(
+        agent_framework=agent_framework,
+        agent_config=search_agent_cfg,
+        tracing=TracingConfig(console=False, cost_info=True),
+    )
+
+    main_agent = AnyAgent.create(
+        agent_framework=agent_framework,
+        agent_config=main_agent_cfg,
+        tracing=TracingConfig(console=False, cost_info=True),
+    )
+
+    serving_config = ServingConfig(
+        port = search_agent_port,
+        endpoint = f"/{search_agent_endpoint}"
+    )
+    thread = Thread(target=lambda: search_agent.serve(serving_config))
+    thread.setDaemon(True)
+    try:
+        thread.start()
+        while not search_agent._server or not search_agent._server.get_uvicorn() or not search_agent._server.get_uvicorn().started:
```

Comment:
since we are extending the class, why not add a method like `_server.ready()`  instead of these checks?

Reply:
After the previous discussions, I didn't know if you wanted to keep the running infra details (for example, the thread supporting the server, whether the serving functions are async or not, etc) in the server class, so I have kept them in the test class itself. https://github.com/mozilla-ai/any-agent/pull/258 has an async server with lifetime control integrated in the server class itself (to be refactored a bit with your latest comments), which I prefer to a threaded implementation. If it's ok, I'll take that approach in this PR too.

Reply:
I'll keep the threaded implementation right now since it seems it's easier to integrate right now. Let's discuss async/threads when the official a2a sdk is used.